File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/review/ReviewResponse.java
package org.camphub.be_camphub.dto.response.review;

import java.util.List;
import java.util.UUID;

import org.camphub.be_camphub.dto.response.MediaResourceResponse;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ReviewResponse {
    UUID id;
    UUID bookingId;
    UUID reviewerId;
    UUID reviewedId;

    String reviewerName;
    String reviewerAvatar;
    String itemName;

    Integer rating;
    String comment;
    List<MediaResourceResponse> mediaUrls;
    String createdAt;
}

File: fe_camphub/src/libs/core/dto/response/review.response.ts
import { MediaResource } from "../../types";

export interface ReviewResponse {
    id: string;
    bookingId: string;
    reviewerId: string;
    reviewedId: string;
    itemName: string;
    reviewerName: string;
    rating: number;
    comment: string;
    mediaUrls: MediaResource[];
    createdAt: string;
}
